### PR TITLE
Minichallenge 2

### PR DESCRIPTION
**Pull request by maxHernandezCastillo**
## Description
```
Added several test to capstone project components.
```
## Type of Change
```
- [x] Added several tests
- [x] AuthContext is used now to store user favorite videos and info(username, avatar).
- [x] Supports now login using username `wizeline` and password `Rocks!`.
```
## How should this be tested?
```
The project can be tested, first installing the necessary packages with `yarn install` and run the test suite using the command `yarn run test:coverage` witch shows the tested components, including the global coverage.
```
## Any background context you want to provide?
```
The app also supports mobile, by being UI responsive.
```
## What is the deliverable?
```
Second deliverable.
```
## Screenshots and links

Screen Shot 2022-01-14 at 17 46 30

### CODE REVIEW

File: src/components/BreadCrumb/BreadCrumb.component.jsx [hunk 1]
```diff
@@ -7,7 +7,10 @@ function BreadCrumb(props) {
   const { authenticated } = useAuth();
 
   return (
-    <ul className={'breadcrumb ' + props.className}>
+    <ul
+      data-testid={props['data-testid']}
```
Comment (jonathanrodgz) [inline src/components/BreadCrumb/BreadCrumb.component.jsx:11]: Great! Using `data-testid`  is better than using `id` or `classname` as selectors, also, there is a way to get rid of these attributes if you don't want to have them in production https://dpericich.medium.com/removing-data-test-attributes-from-react-production-dom-5ea4ea018acc

File: src/components/Header/__test__/Header.test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
+
+import Header from '@components/Header';
+
+const setIsSideBarOpen = jest.fn();
```
Comment (jonathanrodgz) [inline src/components/Header/__test__/Header.test.js:6]: Great, jest functions are really helpful to assert function calls.

File: src/components/LoginModal/LoginModal.component.jsx [hunk 3]
```diff
@@ -7,18 +7,20 @@ import Button from '@components/Button';
 
 function LoginModal(props) {
   const { login } = useAuth();
-  const [username, setUsername] = useState('');
-  const [password, setPassword] = useState('');
+  const [username, setUsername] = useState('wizeline');
+  const [password, setPassword] = useState('Rocks!');
```
Comment (jonathanrodgz) [inline src/components/LoginModal/LoginModal.component.jsx:11]: Did we move test credentials as default values to log in faster by clicking the sign-in button?

File: src/components/LoginModal/__test__/LoginModal.test.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  it('should render', () => {
+    expect(screen.getByTestId('login-modal-component')).toBeInTheDocument();
+  });
+
+  it('should change username input on typing', () => {
+    fireEvent.change(screen.getByLabelText(/.*username.*/i), {
+      target: { value: 'wizeline' },
+    });
+    expect(screen.getByLabelText(/.*username.*/i)).toHaveValue('wizeline');
+  });
```
Comment (jonathanrodgz) [inline src/components/LoginModal/__test__/LoginModal.test.js:27]: 👍

File: src/components/SearchBar/__test__/SearchBar.test.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    userEvent.type(screen.getByPlaceholderText(/.*search.*/i), 'typing-test');
+    expect(screen.getByPlaceholderText(/.*search.*/i)).toHaveValue(
+      'typing-test'
+    );
+  });
+
+  it('should call fetch when clicking typing enter', () => {
+    global.fetch = jest.fn();
+    expect(global.fetch).toBeCalledTimes(1);
+  });
```
Comment (jonathanrodgz) [inline src/components/SearchBar/__test__/SearchBar.test.js:27]: 👍

File: src/components/Sidebar/__test__/Sidebar.test.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    it('should render user image', () => {
+      expect(screen.getByAltText(/.*avatar.*/i)).toBeInTheDocument();
+    });
+    it('should render user username', () => {
+      expect(screen.getByText(/test-username/i)).toBeInTheDocument();
+    });
+    it('should home button redirect to home page', () => {});
+    it('should show logout modal when clicking logout button', () => {});
+    it('should favorites button redirect to favorites page', () => {});
+    it('should show favorites and logout buttons', () => {});
```
Comment (jonathanrodgz) [inline src/components/Sidebar/__test__/Sidebar.test.js:39]: You can use `it.todo` for these tests
https://jestjs.io/docs/api#testtodoname

**Review by jonathanrodgz — APPROVED** [review]
Tests are looking great! This PR can be merged into main 👍

File: src/components/SearchBar/__test__/SearchBar.test.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+  it('should change input value when typing', () => {
+    userEvent.type(screen.getByPlaceholderText(/.*search.*/i), 'typing-test');
+    expect(screen.getByPlaceholderText(/.*search.*/i)).toHaveValue(
+      'typing-test'
+    );
+  });
+
+  it('should call fetch when clicking typing enter', () => {
+    global.fetch = jest.fn();
+    expect(global.fetch).toBeCalledTimes(1);
```
Comment (jonathanrodgz) [inline src/components/SearchBar/__test__/SearchBar.test.js:26]: Is this also failing to you?

Comment (jonathanrodgz) [reply src/components/SearchBar/__test__/SearchBar.test.js:26]: As per our slack conversation this was not implemented at this time, looking for this test to be fixed on the next PRs :)

File: src/components/LoginModal/LoginModal.component.jsx (hunk 3 above)
Comment (maxHernandezCastillo) [reply src/components/LoginModal/LoginModal.component.jsx:11]: Yep due this project is more an exercise, it is just for faster login.

**Pull request merged**